File: Demos/LuaModule/Packets/TestPacket.lua
local Classes = require "Classes"
local ITestPacket = require "ITestPacket"
local BaseType = require "BaseType"

local TestPacket = Classes.class(ITestPacket)

function TestPacket:init()
	self.super:init(self)
end

function TestPacket:FromBinaryReader( reader )
	self.Name.Value = reader:Read()
	self.Password.Value = reader:Read()
end

function TestPacket:ToBinaryWriter( writer )
	writer:Write(BaseType.string,self.Name)
	writer:Write(BaseType.string,self.Password)
end

return TestPacket

File: Demos/PythonModule/Packets/TestPacket.py

from ITestPacket import ITestPacket

class TestPacket(ITestPacket):
	def __init__(self):
		ITestPacket.__init__(self)

	def ToBinaryWriter(self,writer):
		writer.Write(self.string,self.Name)
		writer.Write(self.string,self.Password)

	def FromBinaryReader(self,reader):
		self.Name = reader.Read()
		self.Password = reader.Read()